apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/HashValidatingDownloaderTest.cs

```diff
@@ -36,9 +36,76 @@ public class HashValidatingDownloaderTest
         [InlineData(Md5Value + "," + Crc32Value)]
         [InlineData(Crc32Value + "," + Md5Value)]
         [InlineData(Crc32Value)]
-        public void Valid(string headerValue)
+        public void Valid_ModeAlways(string headerValue)
+        {
+            HashValidatingDownloader downloader = CreateDownloader(request => CreateResponse(request, headerValue), DownloadValidationMode.Always);
+            AssertDownloadSucceeds(downloader);
+        }
+
+        [Theory]
+        [InlineData("gzip", "gzip")]
+        [InlineData(null, "gzip")]
+        [InlineData("plain", null)]
+        [InlineData("plain", "gzip")]
+        [InlineData("gzip", "plain")]
+        [InlineData("gzip", null)]
```

Comment:
Would it be helpful to have 
```
[InlineData(null, null)]
[InlineData("plain","plain")]
```
or are these invalid?

Reply:
I don't think either would actually happen, but I'm happy to add them. (And to InvalidData_ModeAutomatic_Validated as well.)